pkg/storage/fs/owncloud/upload.go

```diff
@@ -81,7 +81,7 @@ func (fs *ocfs) Upload(ctx context.Context, ref *provider.Reference, r io.ReadCl
 
 // InitiateUpload returns an upload id that can be used for uploads with tus
 // TODO read optional content for small files in this request
-func (fs *ocfs) InitiateUpload(ctx context.Context, ref *provider.Reference, uploadLength int64) (uploadID string, err error) {
+func (fs *ocfs) InitiateUpload(ctx context.Context, ref *provider.Reference, uploadLength int64, metadata map[string]string) (uploadID string, err error) {
```

Comment:
Is it intended a `map[string]string`? I think having `map[string]interface{}` is more flexible

Reply:
This is by intention to be aligned with the ArbitraryMetadata approach which has string/string.
At some point we can allow more metadata like checksum and also arbitrary webdav properties